cfn-resources/private-endpoint/cmd/resource/resource.go

```diff
@@ -241,19 +239,25 @@ func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		if epr != nil {
 			return *epr, nil
 		}
-
 		epr = awsvpcendpoint.Delete(req, privateEndpoint.InterfaceEndpoints, *currentModel.Region)
 		if epr != nil {
 			return *epr, nil
 		}
 	} else {
-		response, err = mongodbClient.PrivateEndpoints.Delete(context.Background(), *currentModel.GroupId,
+		_, response, err = mongodbClient.AtlasV2.PrivateEndpointServicesApi.DeletePrivateEndpointService(context.Background(), *currentModel.GroupId,
 			providerName,
-			*currentModel.Id)
+			*currentModel.Id).Execute()
 
 		if err != nil {
-			return progress_events.GetFailedEventByResponse(fmt.Sprintf("Error getting resource : %s", err.Error()),
-				response.Response), nil
+			if response != nil {
+				return progressevent.GetFailedEventByResponse(fmt.Sprintf("Error getting resource : %s", err.Error()),
+					response), nil
+			}
+
+			return handler.ProgressEvent{
+				OperationStatus:  handler.Failed,
+				Message:          err.Error(),
+				HandlerErrorCode: cloudformation.HandlerErrorCodeHandlerInternalFailure}, nil
```

Comment:
Was curious about this check, as I see we usually pass the response pointer to `GetFailedEventByResponse` directly.
As it does give some additional safety, we could consider adding the `response != nil` check directly into the `GetFailedEventByResponse` to avoid duplication everywhere.

Reply:
it makes sense, I will update it. Thanks!